chia/data_layer/data_layer.py

```diff
@@ -264,6 +264,48 @@ async def batch_update(
         else:
             return None
 
+    async def _get_publishable_root_hash(self, tree_id: bytes32) -> bytes32:
+        pending_root: Optional[Root] = await self.data_store.get_pending_root(tree_id=tree_id)
+        if pending_root is None:
+            raise Exception("Latest root is already confirmed.")
+        if pending_root.status == Status.PENDING_BATCH:
+            raise Exception("Unable to publish on chain, batch update set still open.")
+
+        return self.none_bytes if pending_root.node_hash is None else pending_root.node_hash
+
+    async def multistore_batch_update(
+        self,
+        changelist: List[Dict[str, Any]],
+        fee: uint64,
+        submit_on_chain: bool = True,
+    ) -> List[TransactionRecord]:
+        changelist_by_store: Dict[bytes32, List[Dict[str, Any]]] = {}
+        for action in changelist:
+            if "tree_id" not in action:
+                raise Exception("Each operation must have a tree_id field.")
```

Comment:
Was there already discussion around this pattern of a list of items specifying their tree vs. a mapping of trees to lists of items?

Reply:
no, I just picked this as a starting point - looks to me much easier for the user than forcing the grouping in the CLI/RPC.